lib/trip_dispatcher.rb

```diff
@@ -27,14 +33,68 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def available_drivers
+      available_drivers_list = []
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE && (driver.trips.each { |trip| trip.end_time != nil })
```

Comment:
You almost certainly want `all?` here not `each`.  (`each` always returns the original array, `all?` returns a boolean.)

```suggestion
        if driver.status == :AVAILABLE && (driver.trips.all? { |trip| trip.end_time != nil })
```